src/datepicker/datepicker.component.ts

```diff
@@ -269,13 +269,6 @@ export class DatePicker implements
 			// This makes sure that the `flatpickrInstance selectedDates` are in sync with the values of
 			// the inputs when the calendar closes.
 			if (this.range && this.flatpickrInstance) {
-				if (this.flatpickrInstance.selectedDates.length !== 2) {
-					// we could `this.flatpickrInstance.clear()` but it insists on opening the second picker
-					// in some cases, so instead we do this
-					this.setDateValues([]);
```

Comment:
Perhaps, if only 1 date is set of the 2, we should set the start and end date to be the same? This is the behavior I'm seeing in React.

https://react.carbondesignsystem.com/?path=/story/components-datepicker--range-with-calendar

1. Click on the date picker
2. Select a date
3. Click outside of the date picker and notice that the same date is assigned to the start/end range.

Reply:
I update condition part. now it will look like this.

https://github.com/carbon-design-system/carbon-components-angular/assets/49565062/ab959005-a1cc-484a-87f0-293bc96ada01